src/topics/engine/rules/extraction-rules/__tests__/substance/table-extraction.service.test.ts

```diff
@@ -0,0 +1,307 @@
+import type { SpyInstance } from 'vitest';
+import { beforeEach, afterEach, describe, expect, it, vi } from 'vitest';
+
+import type { ILine, IStroke, IText } from '@topics/engine/model/fds.model.js';
+import { aHorizontalStroke, aStroke, aVerticalStroke } from '@topics/engine/__fixtures__/stroke.mother.js';
+import { TableExtractionService } from '@topics/engine/rules/extraction-rules/substance/table-extraction.service.js';
+import { aPosition, aPositionWithXIncremented } from '@topics/engine/__fixtures__/position.mother.js';
+import { POSITION_PROPORTION_X, POSITION_PROPORTION_Y } from '@topics/engine/__fixtures__/fixtures.constants.js';
+import { aLine, aLineWithOneText } from '@topics/engine/__fixtures__/line.mother.js';
+import { aText, aTextWithRandomContent1, aTextWithRandomContent2, aTextWithRandomContent3 } from '@topics/engine/__fixtures__/text.mother.js';
+
+describe('TableExtractionService tests', () => {
+  describe('Vertical stroke tests', () => {
+    describe('mergeStrokesVerticallyAligned tests', () => {
+      it.each<{ message: string; strokes: IStroke[]; expected: IStroke[] }>([
+        {
+          message: 'should return the given stroke if there is only one stroke',
+          strokes: [aStroke().properties],
+          expected: [aStroke().properties],
+        },
+        {
+          message: 'should return all the strokes if they are not on the same x position',
+          strokes: [aStroke().properties, aStroke().withStartBox(aPositionWithXIncremented().properties).properties],
+          expected: [aStroke().properties, aStroke().withStartBox(aPositionWithXIncremented().properties).properties],
+        },
+        {
+          message: 'should return only a stroke if they are on the same x position',
```

Comment:
```suggestion
          message: 'should return only a stroke if they are on the same y position',
```

J'en suis pas certain si tu voulais pas dire y ici ?